channel.go

```diff
@@ -217,6 +217,43 @@ func (c *ChannelService) UpdateChannelMemberNotifications(channelID, userID stri
 	return channelMember, normalizeAppErr(appErr)
 }
 
+// CreateSidebarCategory creates a new sidebar category for a set of channels.
+//
+// Minimum server version: 5.37
+func (c *ChannelService) CreateSidebarCategory(
+	userID, teamID string, newCategory *model.SidebarCategoryWithChannels) error {
+	category, appErr := c.api.CreateChannelSidebarCategory(userID, teamID, newCategory)
+	if appErr != nil {
+		return normalizeAppErr(appErr)
+	}
+	*newCategory = *category
+
+	return nil
+}
+
+// GetSidebarCategories returns sidebar categories.
+//
+// Minimum server version: 5.37
+func (c *ChannelService) GetSidebarCategories(userID, teamID string) (*model.OrderedSidebarCategories, error) {
+	categories, appErr := c.api.GetChannelSidebarCategories(userID, teamID)
+
+	return categories, normalizeAppErr(appErr)
+}
+
+// UpdateSidebarCategories updates the channel sidebar categories.
+//
+// Minimum server version: 5.37
+func (c *ChannelService) UpdateSidebarCategories(
+	userID, teamID string, categories *[]*model.SidebarCategoryWithChannels) error {
+	updatedCategories, appErr := c.api.UpdateChannelSidebarCategories(userID, teamID, *categories)
+	if appErr != nil {
+		return normalizeAppErr(appErr)
+	}
+	*categories = updatedCategories
+
+	return nil
```

Comment:
We can probably remove the pointer here, as the slice is already a pointer, and simply copy the contents. Thoughts, @hanzei?

```suggestion
func (c *ChannelService) UpdateSidebarCategories(
	userID, teamID string, categories []*model.SidebarCategoryWithChannels) error {
	updatedCategories, appErr := c.api.UpdateChannelSidebarCategories(userID, teamID, categories)
	if appErr != nil {
		return normalizeAppErr(appErr)
	}
	copy(categories, updatedCategories)

	return nil
```

Reply:
looks good, but can we be sure that the length of `categories` and `updatedCategories` is always the same?

Reply:
Right now, it is. You can check the code in the channel store here: https://github.com/mattermost/mattermost-server/blob/master/store/sqlstore/channel_store_categories.go#L747-L748 I think it's safe enough to assume it will be like this in the foreseeable future; otherwise, it would be a breaking change.